src/test/programs/federal/snap/unit.test.js

```diff
@@ -535,13 +543,257 @@ describe('SNAPhelpers', () => {
+  describe('`.getRawShelterDeduction( timeClient )` given a time-restricted client object with', () => {
+    let current, getTotalshelterCost, getHalfAdjustedIncome;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+      getTotalshelterCost = jest.spyOn(SNAPhelpers, 'getTotalshelterCost');
+      getHalfAdjustedIncome = jest.spyOn(SNAPhelpers, 'getHalfAdjustedIncome');
+    });
+    afterEach(() => {
+      getTotalshelterCost.mockRestore();
+      getHalfAdjustedIncome.mockRestore();
+    });
+
+    it('returns the shelter cost less the partially adjusted income', () => {
+      const shelterCost = 100;
+      const adjustedIncome = 1;
+      getTotalshelterCost.mockReturnValue(shelterCost);
+      getHalfAdjustedIncome.mockReturnValue(adjustedIncome);
+
+      const rawDeduction = shelterCost - adjustedIncome;
+      expect(SNAPhelpers.getRawShelterDeduction( current )).toEqual(rawDeduction);
+    });
+
+    it('that has a negative raw shelter deduction, it should return zero', () => {
+      const shelterCost = 1;
+      const adjustedIncome = 100;
+      getTotalshelterCost.mockReturnValue(shelterCost);
+      getHalfAdjustedIncome.mockReturnValue(adjustedIncome);
+
+      const rawDeduction = shelterCost - adjustedIncome;
+      expect(rawDeduction).toBeLessThan(0);
+      expect(SNAPhelpers.getRawShelterDeduction( current )).toEqual(0);
+    });
+  });
+
+
   // `SNAPhelpers.getShelterDeduction()`
+  describe('`.getShelterDeduction( timeClient )` given a time-restricted client object with', () => {
+    let current, getRawShelterDeduction;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+      getRawShelterDeduction = jest.spyOn(SNAPhelpers, 'getRawShelterDeduction');
+    });
+    afterEach(() => {
+      getRawShelterDeduction.mockRestore();
+    });
+
+    it('that has an elderly or disabled member, it should return the raw deduction', () => {
+      current.household.push({ m_age: 65, m_disabled: true, m_role: 'member' });
+      const rawDeduction = 12;
+      getRawShelterDeduction.mockReturnValue(rawDeduction);
+      expect(SNAPhelpers.getShelterDeduction( current )).toEqual(rawDeduction);
+    });
+
+    it('that has a raw deduction under the cap, it should return the raw deduction', () => {
+      const rawDeduction = SNAPData.SHELTER_DEDUCTION_CAP - 1;
+      getRawShelterDeduction.mockReturnValue(rawDeduction);
+      expect(SNAPhelpers.getShelterDeduction( current )).toEqual(rawDeduction);
+    });
+
+    it('that has a raw deduction above the cap, it should return the cap', () => {
+      const rawDeduction = SNAPData.SHELTER_DEDUCTION_CAP + 1;
+      getRawShelterDeduction.mockReturnValue(rawDeduction);
+      expect(SNAPhelpers.getShelterDeduction( current )).toEqual(SNAPData.SHELTER_DEDUCTION_CAP);
+    });
+  });
+
+
   // `SNAPhelpers.getHomelessDeduction()`
+  describe('`.getHomelessDeduction() given a time-restriced client object with', () => {
```

Comment:
Again, if you get a chance, change 'with' to 'that is' and remove it from the `it()` descriptions.